src/core_ocean/analysis_members/mpas_ocn_mixed_layer_heat_budget.F

```diff
@@ -0,0 +1,417 @@
+         call mpas_pool_get_array(diagnosticsPool, 'BruntVaisalaFreqTop', BruntVaisalaFreqTop)
+
+         call mpas_pool_get_array(statePool, 'layerThickness', layerThickness)
+
+         call mpas_pool_get_array(mixedLayerHeatBudgetAMPool, 'activeTracerForcingMLTend', &
+                                  activeTracerForcingMLTend)
+         call mpas_pool_get_array(mixedLayerHeatBudgetAMPool, 'activeTracerNonLocalMLTend', &
+                                  activeTracerNonLocalMLTend)
+         call mpas_pool_get_array(mixedLayerHeatBudgetAMPool, 'activeTracerHorMixMLTend', &
+                                  activeTracerHorMixMLTend)
+         call mpas_pool_get_array(mixedLayerHeatBudgetAMPool, 'activeTracerVertMixMLTend', &
+                                  activeTracerVertMixMLTend)
+         call mpas_pool_get_array(mixedLayerHeatBudgetAMPool, 'activeTracerHorAdvectionMLTend', &
+                                  activeTracerHorAdvectionMLTend)
+         call mpas_pool_get_array(mixedLayerHeatBudgetAMPool, 'activeTracerVertAdvectionMLTend', &
+                                  activeTracerVertAdvectionMLTend)
+         call mpas_pool_get_array(mixedLayerHeatBudgetAMPool, 'activeTracersML', activeTracersML)
+         call mpas_pool_get_array(mixedLayerHeatBudgetAMPool, 'bruntVaisalaFreqML', bruntVaisalaFreqML)
+         call mpas_pool_get_array(mixedLayerDepthsAMPool, 'dThreshMLD', dThreshMLD)
+
+         call mpas_pool_get_array(tracersPool, 'activeTracers', tracers, 1)
+
+         num_tracers = size(activeTracerVertMixTendency, dim=1)
+
+         do iCell = 1, nCellsSolve
+            activeTracerHorAdvectionMLTend(:, iCell) = 0.0_RKIND
+            activeTracerNonLocalMLTend(:, iCell) = 0.0_RKIND
+            activeTracerHorMixMLTend(:, iCell) = 0.0_RKIND
+            activeTracerVertMixMLTend(:, iCell) = 0.0_RKIND
+            activeTracerVertAdvectionMLTend(:, iCell) = 0.0_RKIND
+            activeTracerForcingMLTend(:, iCell) = 0.0_RKIND
+            activeTracersML(:, iCell) = 0.0_RKIND
+            k = 1
+            depth = layerThickness(k, iCell)
+            do while (depth <= dThreshMLD(iCell) .and. k < maxLevelCell(iCell))
+               do iTracer = 1, num_tracers
+                  activeTracerHorAdvectionMLTend(iTracer, iCell) = activeTracerHorAdvectionMLTend(iTracer, iCell) + &
+                                                 activeTracerHorizontalAdvectionTendency(iTracer, k, iCell)*layerThickness(k, iCell)
+                  activeTracerHorMixMLTend(iTracer, iCell) = activeTracerHorMixMLTend(iTracer, iCell) + &
+                                                             activeTracerHorMixTendency(iTracer, k, iCell)*layerThickness(k, iCell)
+                  activeTracerForcingMLTend(iTracer, iCell) = activeTracerForcingMLTend(iTracer, iCell) + &
+                                                         activeTracerSurfaceFluxTendency(iTracer, k, iCell)*layerThickness(k, iCell)
```

Comment:
just a  style comment, feel free to ignore: I'd put the forcing terms following each other, like you did for HorAdv/HorMix, VertAdv/VertMix.